### ISSUE TITLE
`cargo clippy` does not process build scripts

### ISSUE THREAD

**Issue by FaultyRAM** [rust-lang/rust-clippy#2802]
`cargo clippy` doesn't do any extra handling of build scripts beyond what `cargo check` does. For example, this code will trigger an `unknown_lints` warning when using `cargo clippy`:

```rust
#![forbid(print_literal)]

fn main() {
    println!("{}", "");
}

```

Built-in lints still work; adding a `#![forbid(missing_docs)]` to the above snippet fails the build as expected. clippy-as-a-plugin also works:

```rust
#![feature(plugin)]
#![plugin(clippy)]
#![forbid(print_literal)]

fn main() {
    println!("{}", "");
}

```

I'm not actually sure whether clippy or cargo is at fault here, but since the plugin interface is about to go I figured it might be more prudent to report here first.

**Comment by lopopolo** [comment]
@flip1995 I noticed this issue has no tags. Does that mean it is not prioritized?

**Comment by Manishearth** [comment]
No. Not all issues fit cleanly in the labeled buckets.

This in particular is something that would require https://github.com/rust-lang/rust-clippy/issues/3837 to be finished

**Comment by nk9** [comment]
https://github.com/rust-lang/rust/issues/44690 is complete, and #3837 was closed as WONTFIX. Can this issue be closed now?

@rustbot label S-triage
